js/client.js

```diff
@@ -869,6 +869,24 @@
 			}
 
 			switch (parts[0]) {
+			case 'authconfig':
+				var nlIndex = data.indexOf('\n');
+				if (nlIndex > 0) {
+					this.receive(data.substr(nlIndex + 1));
+				}
+
+				var tarRow = data.slice(12, nlIndex).split(/(?:\b|\B)\|(?:\B)/);
+				for (var i = 0; i < tarRow.length; i++) {
+					var entry = tarRow[i];
+					if (!entry) continue;
+
+					var symbol = entry.charAt(0);
+					var rank = entry.slice(1);
+
+					groupDetails[symbol] = rank + ' (' + symbol + ')'; // override the default symbol.
```

Comment:
We should probably do `delete groupDetails[symbol];` first, to support reordering hardcoded ranks.